From 18d5e0e0260a50e2a466278e45aa7719f1f7815c Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Tue, 8 Sep 2015 14:17:23 +0800
Subject: [PATCH 0532/5242] MLK-11503-1 mmc: sdhci-pltfm: add pinctrl sleep
 mode support

commit  81898e7ba1f3e4f0b0fa4b2d373493f24f231f55 from
https://source.codeaurora.org/external/imx/linux-imx.git

For LPSR mode, usdhc iomux settings will be lost after resume,
so add pinctrl sleep mode support.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
(cherry picked from commit 983a7a174ed20d34a170a6aba70ff9d5bb2c9973)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mmc/host/sdhci-pltfm.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-pltfm.c b/drivers/mmc/host/sdhci-pltfm.c
index 02bea61..659b1c7 100644
--- a/drivers/mmc/host/sdhci-pltfm.c
+++ b/drivers/mmc/host/sdhci-pltfm.c
@@ -2,7 +2,7 @@
  * sdhci-pltfm.c Support for SDHCI platform devices
  * Copyright (c) 2009 Intel Corporation
  *
- * Copyright (c) 2007, 2011 Freescale Semiconductor, Inc.
+ * Copyright (C) 2007, 2011, 2015 Freescale Semiconductor, Inc.
  * Copyright (c) 2009 MontaVista Software, Inc.
  *
  * Authors: Xiaobo Xie <X.Xie@freescale.com>
@@ -224,6 +224,8 @@ int sdhci_pltfm_suspend(struct device *dev)
 
 	clk_disable_unprepare(pltfm_host->clk);
 
+	pinctrl_pm_select_sleep_state(dev);
+
 	return 0;
 }
 EXPORT_SYMBOL_GPL(sdhci_pltfm_suspend);
@@ -234,6 +236,8 @@ int sdhci_pltfm_resume(struct device *dev)
 	struct sdhci_pltfm_host *pltfm_host = sdhci_priv(host);
 	int ret;
 
+	pinctrl_pm_select_default_state(dev);
+
 	ret = clk_prepare_enable(pltfm_host->clk);
 	if (ret)
 		return ret;
-- 
1.7.9.5

